riscv/csrs.cc

```diff
@@ -1323,8 +1323,48 @@ fcsr_csr_t::fcsr_csr_t(processor_t* const proc, const reg_t addr, csr_t_p upper_
   composite_csr_t(proc, addr, upper_csr, lower_csr, upper_lsb) {
 }
 
+void fcsr_csr_t::verify_permissions(insn_t insn, bool write) const {
+  if (proc->extension_enabled(EXT_SMSTATEEN) && proc->extension_enabled(EXT_ZFINX)) {
+    if ((state->prv < PRV_M) && !(state->mstateen[0]->read() & MSTATEEN0_FCSR))
+      throw trap_illegal_instruction(insn.bits());
+
+    if (state->v && !(state->hstateen[0]->read() & HSTATEEN0_FCSR))
+      throw trap_virtual_instruction(insn.bits());
+
+    if ((proc->extension_enabled('S') && state->prv < PRV_S) && !(state->sstateen[0]->read() & HSTATEEN0_FCSR)) {
```

Comment:
I believe Smstateen requires S-mode, yes? If so the `extension_enabled()` check here is unnecessary.

Reply:
Sorry. I didn't find this in the spec. I'll check this later.

Reply:
I think it doesn't require S-mode, since in chapter 1 of the spec, it have a description for sstateen csrs: "If supervisor mode is implemented, another four CSRs are defined at supervisor level"